File: Bisection.java
import java.util.*;
class Bisection
{
	static float f (float x)
	{
		return ((x*x)-(4*x)+4);
	}
	public static void main(String args[])
	{
		int i=1;
		float a,b,m;
		double e=0.0001;
		System.out.println("\n Enter the value for a & b");
		Scanner in=new Scanner(System.in);
		System.out.println("Enter the value of a and b");
		a=in.nextFloat();
		b=in.nextFloat();
		if((f(a)*f(b))>0)
			System.out.println("Interval is invalid");
		else
		{
			m=(a+b)/2;
			while(Math.abs(f(m))>=0)
			{
				if(f(m)==0)
				{
					System.out.println("root found!");
					break;
				}
			
			    else
			    {
				   System.out.println(a+" "+b+" "+i+" "+f(m)+" "+m);
				    if((f(a)*f(m))>0)
				      a=m;
				    else
					  b=m;
				    m=(a+b)/2;
				    i=i+1;
			    }
		    }
		    System.out.println(a+" "+b+" "+i+" "+f(m)+" "+m);
		    System.out.println("Root is "+m);
		}
		
		
	}
}

File: Bisection.R
func <- function(x)
{
  value <- x*x*x-4*x-9
  return(value)
}
i <- 1
e <- 0.0001
print ("Enter a value for a & b ")
a <- as.double(readline(prompt="a :" ))
b<- as.double(readline(prompt=" b : "))
if(func(a)*func(b)>0)
{
print ("Interval invalid")
}else
{
  m <- (a+b)/2
  while(((b-a)/b)>0.0001)
  {
      if(func(m)==0.0001)
      {
          print ("Root Found !")
          break 
      }else
      {
          print(paste(a , b , i , func(m), m))
          if((func(a)*func(m))>0)
                { a=m}else{b=m}
          m= (a+b)/2
          i=i+1
      }
  } 
  print(paste(" Root is :",m))
}